src/main/java/com/currencyexchange/controller/CurrencyController.java

```diff
@@ -86,16 +90,18 @@ public class CurrencyController {
                     schema =
                         @Schema(
                             type = "string",
-                            example = "Internal server error.",
+                            example =
+                                "{\"error\": \"Unauthorized\", "
+                                    + "\"message\": \"Internal server error\"}",
                             description = "Error message when server encounters an issue")))
       })
   @ResponseStatus(HttpStatus.OK)
   @GetMapping("/")
-  public Set<String> getAllCurrencies() {
+  public CurrencyListDto getAllCurrencies() {
```

Comment:
It should be CurrencyDto with one currency if you want to use DTO. For CurrencyListDto you should use Set<CurrencyDto>. 
Actually we need model for currency here instead of DTO